script.js

```diff
@@ -1,49 +1,64 @@
-
-let apiKey = "&key=AIzaSyCw6aeDwd2srAlaiVBFSSBvyl2O7atCTaA";
+const googleBooksURL = "https://www.googleapis.com/books/v1/volumes?q=";
+const apiKey = "&key=AIzaSyCw6aeDwd2srAlaiVBFSSBvyl2O7atCTaA";
 
 searchForm.addEventListener("submit", () => {
   msgWaiting.style.display = "none";
-  let searchInput = document.getElementById("searchInput").value;
-  let searchResult = `${getGoogleBooks}${searchInput}${apiKey}`;
-
-  async function getBooks() {
-    const booksPromise = await fetch(searchResult);
-    const getBook = await booksPromise.json();
-    booksDisplay.innerHTML = "";
-    getBook.items.map((book) => {
-      console.log(book.volumeInfo);
-
-      let title = book.volumeInfo.title;
-      let authors = book.volumeInfo.authors;
-      let previewLink = book.volumeInfo.previewLink;
-      let publishdate = book.volumeInfo.publishedDate;
-      let publisher = book.volumeInfo.publisher;
-      let smallThumbnail = book.volumeInfo.imageLinks.smallThumbnail;
-
-      let div = document.createElement("div");
-      div.innerHTML = `
-      
-      <h3>  ${title} </h3>
-      <div class= "book-img">
-       <img src= "${smallThumbnail}" alt= "Image book cover "> 
-       </div>
-        <div class = "book-details">
+  const searchInput = document.getElementById("search-input").value;
+  const searchQuery = `${googleBooksURL}${searchInput}${apiKey}`;
+
+  getBooks(searchQuery);
+});
+
+async function getBooks(query) {
+  const booksPromise = await fetch(query);
+  const booksData = await booksPromise.json();
+
+  displayBooks(booksData.items);
+}
+
+function displayBooks(books) {
+  // clear any previous results
+  booksDisplay.innerHTML = "";
+
+  books.forEach((book) => {
+
+    let volumeInfo = book.volumeInfo;
+    let title = volumeInfo.title || "N/A";
+    let authors = volumeInfo.authors || "N/A";
+    let previewLink = volumeInfo.previewLink || "N/A";
+    let publishdate = volumeInfo.publishedDate || "N/A";
+    let publisher = volumeInfo.publisher || "N/A";
+    let smallThumbnail = getThumbnail(volumeInfo);
+
+    let bookDiv = document.createElement("div");
+
+    bookDiv.classList.add("book-styling");
+    bookDiv.innerHTML = `
+    <h3>  ${title} </h3>
+    <div class= "book-img">
+     <img src= "${smallThumbnail}" alt= "Image book cover ">
+     </div>
+      <div class = "book-details">
         <small> <strong>Author(s)</strong>: ${authors}</small>
         <small> <strong>Publishers</strong>: ${publisher}</small>
         <small> <strong>Published</strong>: ${publishdate}</small>
-        </div>
-        <a href = "${previewLink}" class ="book-link" target="_blank">  Google Books Preview </a>
-         `;
-      div.classList.add("book-styling");
-
-      booksDisplay.append(div);
-    });
-  }
-  getBooks();
-});
+      </div>
+      <a href = "${previewLink}" class ="book-link" target="_blank">  Google Books Preview </a>
+       `;
+
+    booksDisplay.append(bookDiv);
+  });
+}
+
+function getThumbnail(volumeInfo) {
```

Comment:
Here, we need to return a string with the URL to the book's thumbnail image. However, not all books have one.

This means we need to first check whether the returned search result has `imageLinks` and a `smallThumbnail` field. If it does, we can return the `smallThumbnail` field.

If it doesn't, there are several options. Because I didn't wanna look for a more appropriate thumbnail (like, for example, a generic book icon), I just return a Picsum (like Lorem Ipsum but for images). This will return a random image.

I don't actually like this solution because you have no idea what that random image is going to be, and I wouldn't particularly trust Picsum not to return something offensive or inappropriate. So I suggest you find a free to use book icon you can use in this situation, or come up with another solution that won't break your design when a search result doesn't have a thumbnail.

Note: For reference, on Jan 10, 2021, the 3rd result when searching for 'Moon of the Crusted Snow' doesn't have a thumbnail.